server/src/connections/redis-dal.ts

```diff
@@ -0,0 +1,78 @@
+import { createClient, RedisClientType } from 'redis';
+import { REDIS_URL } from '../utils/environment-variables';
+
+interface IRedisCacheProps<T> {
+	key: string;
+	callbackFn: () => Promise<T>;
+	expirationTime: number;
+}
+
+interface IRedisSetValueProps<T> {
+	key: string;
+	value: T;
+	expirationTime: number;
+}
+
+export class RedisDAL {
+	public redisClient: RedisClientType;
+	// connect to redis
+	constructor() {
+		if (!this.redisClient) {
+			try {
+				this.redisClient = createClient({ url: REDIS_URL });
```

Comment:
can we do something like if coudnt connect to `REDIS_URL` then connect locally?

Reply:
it's should be handle in util file for environment. Once we are in production we want to connect directly to the container.